## add data to session

flagship_kidgen = Flagship(flagship_name='KidGen Renal genetics', flagship_institute=' Queensland Health', flagship_lead='Andrew Mallett', flagshipDiseaseType='Kidney')
flagship_neuro = Flagship(flagship_name='Neuromuscular disorders', flagship_institute='University of Western Australia', flagship_lead='Nigel Laing', flagshipDiseaseType='Neuromuscular')
flagship_mito = Flagship(flagship_name='Mitochondrial disorders', flagship_institute='Murdoch Childrens Hospital', flagship_lead='John Christodoulou', flagshipDiseaseType='Mitochondrial')
flagship_som = Flagship(flagship_name='Somatic cancer', flagship_institute='Peter MacCallum Cancer Centre', flagship_lead='Stephen Fox', flagshipDiseaseType='Cancer')
flagship_brain = Flagship(flagship_name='Brain malformations & leukodystrophies', flagship_institute='University of Melbourne', flagship_lead='Ingrid Scheffer', flagshipDiseaseType='Epileptic Encephalopathy')
flagship_enc = Flagship(flagship_name='Epileptic Encephalopathy', flagship_institute='Murdoch Childrens Hospital', flagship_lead='Richard Leventer', flagshipDiseaseType='Brain malformations and leukodystrophies')
flagship_imm = Flagship(flagship_name='Genetic Immunology', flagship_institute='Australian Nataional University', flagship_lead='Matthew Cook', flagshipDiseaseType='Immunology')
flagship_mgha = Flagship(flagship_name='Melbourne Genomics Health Alliance', flagship_institute='Melbourne Genomics Health Alliance', flagship_lead='Melbourne Genomics Health Alliance', flagshipDiseaseType='null')
flagship_paed = Flagship(flagship_name='Germline ICCon', flagship_institute='University of Queensland', flagship_lead='Robyn Ward & Gillian Mitchell', flagshipDiseaseType='ICCon')
flagship_null = Flagship(flagship_name='Not affiliated with a Flagship', flagship_institute='null', flagship_lead='null', flagshipDiseaseType='null')

institute_imb = Institute(institute_name = 'Institute for Molecular Bioscience', institute_logo = 'null', institute_long= '153.009784', institute_lat= '-27.498094')
institute_kccg = Institute(institute_name = 'Kinghorn Centre for Clinical Genomics', institute_logo = 'null', institute_long= '151.221706', institute_lat= '-33.878596')
institute_vcgs = Institute(institute_name = 'Victorian Clinical Genetic Services', institute_logo = 'null', institute_long= '144.949343', institute_lat= '-37.793749')
institute_mb = Institute(institute_name = 'Melbourne Bioinformatics', institute_logo = 'null', institute_long= '144.962710', institute_lat= '-37.800365')
institute_pm = Institute(institute_name = 'Peter MacCallum Cancer Centre', institute_logo = 'null', institute_long= '144.956371', institute_lat= '-37.800277')
institute_atcg = Institute(institute_name = 'Institute of Health and Biomedical Innovation', institute_logo = 'null', institute_long= '153.013758', institute_lat= '-27.452918')
institute_sap = Institute(institute_name = 'South Australia pathology', institute_logo = 'null', institute_long= '138.599898', institute_lat= '-34.911586')
institute_qimr = Institute(institute_name = 'QIMR Berghofer', institute_logo = 'null', institute_long= '153.027383', institute_lat= '-27.449517')
institute_pathwest = Institute(institute_name = 'Pathwest', institute_logo = 'null', institute_long= '115.866493', institute_lat= '-31.954350')
institute_schn = Institute(institute_name = 'Sydney Childrens Hospital Network', institute_logo = 'null', institute_long= '151.237574', institute_lat= '-33.917409')
institute_nci = Institute(institute_name = 'National Computational Infrastructure', institute_logo = 'null', institute_long= '149.114928', institute_lat= '-35.282156')


pipeline_imb = Pipeline(pipeline_name='IMB Trio', institute=institute_imb)
pipeline_kccg = Pipeline(pipeline_name='Garvan', institute=institute_kccg)
pipeline_vcgs = Pipeline(pipeline_name='Cpipe', institute=institute_vcgs)
pipeline_mb_cpipe = Pipeline(pipeline_name='Cpipe', institute=institute_mb)
pipeline_pm = Pipeline(pipeline_name='Seqliner', institute=institute_pm)
pipeline_mb_seq = Pipeline(pipeline_name='Seqliner', institute=institute_mb)
pipeline_atcg = Pipeline(pipeline_name='ATCG pipeline', institute=institute_atcg)
pipeline_sap = Pipeline(pipeline_name='SAP pipeline', institute=institute_sap)
pipeline_qimr = Pipeline(pipeline_name='QIMR pipeline', institute=institute_qimr)
pipeline_nci = Pipeline(pipeline_name='NCI pipeline', institute=institute_nci)
pipeline_pathwest = Pipeline(pipeline_name='Pathwest pipeline', institute=institute_pathwest)
pipeline_schn = Pipeline(pipeline_name='Sydney Childrens Hospital Network pipeline', institute=institute_schn)

workflow_pm_panel = Workflow(workflow_name = 'Seqliner (clinical)', library_preparation = 'Comprehensive cancer panel version 2', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'yes', reference_genome = 'GrCh37', workflow_usage = 'diagnostic', workflow_accession='explorer-html/seqliner-clinical-index-v1.html', pipeline=pipeline_pm, flagship=flagship_som)
workflow_pm_exome = Workflow(workflow_name = 'Seqliner (research)', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'GrCh37' , workflow_usage = 'research', workflow_accession='explorer-html/seqliner-t-n-index-v1.html', pipeline=pipeline_pm, flagship=flagship_null)
workflow_mb_exome = Workflow(workflow_name = 'Cpipe lymphoma exome', library_preparation = 'Nextera exome', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/cpipe-index-v1.html', pipeline=pipeline_mb_cpipe, flagship=flagship_mgha)
workflow_vcgs_exome = Workflow(workflow_name = 'Cpipe exome', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no', reference_genome = 'hg19' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/cpipe-index-v1.html', pipeline=pipeline_vcgs, flagship=flagship_brain)
workflow_vcgs_panel1 = Workflow(workflow_name = 'Cpipe panel', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'no', reference_genome = 'hg19' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/cpipe-index-v1.html', pipeline=pipeline_vcgs, flagship=flagship_enc)
workflow_vcgs_panel2 = Workflow(workflow_name = 'Cpipe panel', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'no', reference_genome = 'hg19' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/cpipe-index-v1.html', pipeline=pipeline_vcgs, flagship=flagship_mito)
workflow_sap_exome = Workflow(workflow_name = 'SAP medical exome', library_preparation = 'Nimblegen medical exome', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'hg19 v37 decoy' , workflow_usage = 'diagnostic & research', workflow_accession='explorer-html/sap-exome-index-v1.html', pipeline=pipeline_sap, flagship=flagship_neuro)
workflow_sap_panel = Workflow(workflow_name = 'SAP panel', library_preparation = 'Trusight One Illumina Panel', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'no', reference_genome = 'hg19 v37 decoy' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/sap-exome-index-v1.html', pipeline=pipeline_sap, flagship=flagship_neuro)
workflow_kccg_som = Workflow(workflow_name = 'Garvan somatic', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'no', reference_genome = 'various' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/garvan-germline-index-v1.html', pipeline=pipeline_kccg, flagship=flagship_null)
workflow_kccg_germ = Workflow(workflow_name = 'Garvan germline', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'no', reference_genome = 'b37d5 (1000 genomes + decoy)' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/garvan-germline-index-v1.html', pipeline=pipeline_kccg, flagship=flagship_kidgen)
workflow_qimr_som = Workflow(workflow_name = 'QIMR somatic', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'in completion' , reference_genome = 'GrCh37.75.ICGC.standard_v2 - no decoy' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/qimr-index-v1.html', pipeline=pipeline_qimr, flagship=flagship_null)
workflow_qut_som = Workflow(workflow_name = 'QUT somatic', library_preparation = '75 bp paired end Illumina sequences - IDT exome capture with spike-in', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'in completion' , reference_genome = 'hg19 + decoy' , workflow_usage = 'both', workflow_accession='explorer-html/qut-index-v1.html', pipeline=pipeline_atcg, flagship=flagship_null)
workflow_nci = Workflow(workflow_name = 'NCI germline', library_preparation = 'n/a', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'in completion' , reference_genome = 'Hs37d5 + decoy' , workflow_usage = 'both', workflow_accession='explorer-html/nci-index-v1.html', pipeline=pipeline_nci, flagship=flagship_imm)
workflow_pathwest_panel = Workflow(workflow_name = 'Pathwest panel', library_preparation = 'Trusight muscle, Trusight neuromuscular', library_layout = 'paired', sequencing_strategy = 'Panel', nata_accreditation = 'Yes' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', workflow_accession='explorer-html/pathwest-index-v1.html', pipeline=pipeline_pathwest, flagship=flagship_neuro)
workflow_schn_panel = Workflow(workflow_name = 'SCHN panel', library_preparation = 'Trusight panel', library_layout = 'paired', sequencing_strategy = 'Panel', nata_accreditation = 'Yes' , reference_genome = 'GrCh37' , workflow_usage = 'Diagnostic', workflow_accession='explorer-html/schn-index-v1.html', pipeline=pipeline_schn, flagship=flagship_kidgen)
workflow_imb_trio = Workflow(workflow_name = 'IMB trio germline', library_preparation = 'Nextera Exome', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'No' , reference_genome = 'hg19' , workflow_usage = 'Research', workflow_accession='explorer-html/imb-index-v1.html', pipeline=pipeline_imb, flagship=flagship_kidgen)



def_term1 = Terminology(term_name='Base quality score recalibration', term_definition='Quality scores output by sequencer reflect confidence of the machine in calling a given base. Quality scores are based on sequencer algorithms. Variant callers are heavily reliant on quality scores assigned to each base in each read. BSQR applies machine learning to model errors empirically and adjust quality scores. As the majority of variants have been deposited in dbSNP, novel variants are likely to be sequencing errors. Known variants are masked and the quality scores of remaining SNP’s examined in the context of covariates (quality of surrounding bases, nearby mismatches). ', term_type='data_operation', provenance='GATK')
def_term2 = Terminology(term_name='Coverage', term_definition='Number of reads that cover a given nucleotide position.', term_type='concept', provenance='not listed')
def_term3 = Terminology(term_name='Consequence term', term_definition='As defined by the Sequence ontology. Predict the effect of each allele of each variant on the transcript. Provides a score of impact based on position of variant in the transcript (e.g. splice site, UTR etc).', term_type='ontology_term', provenance='not listed')
def_term4 = Terminology(term_name='Decomposition', term_definition='Breaking variants into multiple records – splitting multi-allelic sites into separate entries', term_type='data_operation', provenance='not listed')
def_term5 = Terminology(term_name='De novo mutations', term_definition='Parents are homozygous reference whereas the child is a het (carries a copy of the alternate allele). A high confidence de novos would be sites where GQ >= 20 across trio with AC <4 or AF <0.1%.', term_type='concept', provenance='not listed')
def_term6 = Terminology(term_name='PCR Duplicates', term_definition='Duplicates are an artefact of the sequencing process and can be defined as reads arising from the same DNA fragment. In order to generate a signal strong enough for the sequencer to detect, fragments need to be amplified using PCR. The lower the amount of starting the material, the more amplification cycles need to be run. They can be identified by reads mapping to the same genomic coordinates. Failure to remove duplicates can lead to skewed allele frequencies. A high rate of duplication can be diagnostic of an enrichment bias. ', term_type='concept', provenance='not listed')
def_term7 = Terminology(term_name='Genotype refinement', term_definition='Calculate the probability a sample takes on a homref, het or homref genotype using sample PL. Variation observed for individuals within the trio is used to improve likelihood estimates. Confidence level for genotypes uses Mendelian inheritance rules to strengthen predictions. De novo mutations are tagged (VariantAnnotator) using posterior genotype probabilities. ', term_type='data_operation', provenance='GATK')
def_term8 = Terminology(term_name='HaplotypeCaller', term_definition='Discovery of germline SNP’s and indels via local re-assembly of haplotypes. Regions of variation are identified and the region is reassembled locally to identify all possible haplotypes. Possible haplotypes are realigned against the reference to calculate most probable variant. Bayes rule is applied using the likelihood of alleles based on the given the reads to determine the most probable genotype. Genotyping algorithm used to detect het vs hom (expects 0, 50, 100% at variant calling sites). NB this is problematic for somatic samples where frequencies are typically skewed (mutation feature, also contamination).', term_type='data_operation', provenance='GATK')
def_term9 = Terminology(term_name='Insert', term_definition='Refers to the read between the paired end adaptors. Insert size should be ~ 300-500 bp for paired end sequencing.  For illumina, this would be the difference between the 5’ positions of the two reads.', term_type='concept', provenance='not listed')
def_term10 = Terminology(term_name='Interval padding', term_definition='Convention of shifting indels furthest left whilst keeping the length of the allele constant. Applies to situations where the indel position can be ambiguous (e.g. STR).', term_type='', provenance='')
def_term11 = Terminology(term_name='Left align', term_definition='Convention of shifting indels furthest left whilst keeping the length of the allele constant. Applies to situations where the indel position can be ambiguous (e.g. STR).', term_type='', provenance='')
def_term12 = Terminology(term_name='Local realignment', term_definition='Serves to reduce detection of false SNP’s generated by incorrect mapping around regions containing indels. Read mapping algorithms operate on each read independently. ‘Hidden’ indels can lead to misalignments leading to discovery of false variants. Local realignment examines all reads which span a position to determine the highest-possible scoring consensus. Sites of high variation are targeted.', term_type='')
def_term13 = Terminology(term_name='Mixed', term_definition="Multinucleotide and an indel at a position e.g. Reference = 'ATA', Sample = 'GTCAGT';", term_type='', provenance='')
def_term14 = Terminology(term_name='MNP', term_definition="multinucleotide variant e.g. Reference = 'ATA', Sample = 'GTC'– substitution mutations across consecutive bases.'", term_type='', provenance='')
def_term15 = Terminology(term_name='Normalisation', term_definition='Describes application of both left alignment and parsimony. see XX', term_type='', provenance='')
def_term16 = Terminology(term_name='Parsimony', term_definition='Representation of a variant by as few nucleotides as possible with allele length > 0. This involves trimming bases on either side of the variant.', term_type='', provenance='')
def_term17 = Terminology(term_name='Proband', term_definition='Person with the genetic disorder (usually the child).', term_type='', provenance='')
def_term18 = Terminology(term_name='Refseq accessions', term_definition='Prefix XM are model refseqs. These represent predicted transcripts produced by an automated annotation pipeline and may differ from curated records. NM represent known transcripts with evidence available and curated.', term_type='', provenance='')
def_term18 = Terminology(term_name='SAM file', term_definition='Tab-delimited text file containing aligned sequence data', term_type='', provenance='')
def_term19 = Terminology(term_name='Symbolic', term_definition='Mutation event not well-modelled', term_type='', provenance='')
def_term20 = Terminology(term_name='Variant annotation', term_definition='Process of ascribing biological function/impact to variant calls', term_type='', provenance='')
def_term21 = Terminology(term_name='GT', term_definition='Genotype at this site, encoded 0 for ref and 1 for alt.', term_type='', provenance='')
def_term22 = Terminology(term_name='AD', term_definition='Unfiltered allele depth (counts) – the number of reads at the position, including those that failed variant caller filters.', term_type='', provenance='')
def_term23 = Terminology(term_name='DP', term_definition='filtered allele depth (counts), including only those which passed variant caller filters.', term_type='', provenance='')
def_term24 = Terminology(term_name='PL', term_definition='Normalised Phred-scaled likelihoods – the likelihood of the genotype scaled by quality', term_type='', provenance='')
def_term25 = Terminology(term_name='GQ', term_definition='Quality of the assigned genotype. Calculated as the difference between two most likely genotypes (based on PL). Low GQ indicates low confidence in choosing one genotype over another.', term_type='', provenance='')
def_term26 = Terminology(term_name='Biotype', term_definition='Biotype of a transcript or regulatory feature. GENCODE term.', term_type='', provenance='GENCODE')
def_term27 = Terminology(term_name='MAF', term_definition='Minor allele frequency, refers to the frequency at which the second most common allele occurs in a given population.', term_type='', provenance='')
def_term28 = Terminology(term_name='Raw coverage', term_definition='Percentage of reads that map to reference genome', term_type='', provenance='')
def_term29 = Terminology(term_name='Informative coverage', term_definition='Uniquely mapped reads ', term_type='', provenance='')
def_term30 = Terminology(term_name='Horizontal coverage', term_definition='Target mapped reads on or near target ', term_type='', provenance='')
def_term31 = Terminology(term_name='Depth', term_definition='Read depth. Vertical coverage. Number of times a given base position has been sequenced', term_type='', provenance='')
def_term32 = Terminology(term_name='Depth of coverage', term_definition='Breadth of coverage. Read length * number of reads / haploid genome length. Higher read depth = more reliable consensus and improved accuracy.', term_type='', provenance='')
def_term33 = Terminology(term_name='Transition/transversion ratios', term_definition=' Serves as a general indicator of quality - ratio should remain constant for targeted regions. Comparable values from literature available.', term_type='', provenance='')
def_term34 = Terminology(term_name='Mapping quality score ', term_definition='probability a read is misplaced based on number of distinct regions within a reference assembly software could map the read, no of base differences between the read and candidate locations in the reference sequence. Mapping scores from different algorithms not comparable. Low mapping quality can be indicative of non-specific amplification, capture off target, contaminatio', term_type='', provenance='')
def_term35 = Terminology(term_name='Capture efficiency', term_definition='percentage of targeted bases that are covered by one or more reads.', term_type='', provenance='')
def_term36 = Terminology(term_name='refseq', term_definition='Genes, transcripts, proteins. Annotated set of reference sequences including genomic, transcript, and protein. A subset of annotations from UCSC and Ensembl.', term_type='database', provenance='NCBI')
def_term37 = Terminology(term_name='refseqgene', term_definition='Subset of NCBI, defines genomic sequences to be used as reference standards for well-characterised genes.', term_type='database', provenance='NCBI')
def_term38 = Terminology(term_name='dbSNP', term_definition='SNP variant database. Contains neutral and disease-causing variants', term_type='database', provenance='NCBI')
def_term39 = Terminology(term_name='dbVAR', term_definition='Structural variant database. contains insertions, deletions, duplications, inversions, multinucleotide substitutions, mobile element insertions, translocations, and complex chromosomal rearrangements. See dbSNP for single nucleotide substitutions', term_type='database', provenance='')
def_term40 = Terminology(term_name='1000 genomes', term_definition='Catologue of human variation', term_type='database', provenance='IGSR')
def_term41 = Terminology(term_name='Genome in a bottle', term_definition='Sets of reference genomes. to develop the technical infrastructure (reference standards, reference methods, and reference data) to enable translation of whole human genome sequencing to clinical practice.', term_type='database', provenance='')
def_term42 = Terminology(term_name='COSMIC', term_definition='Catalogue Of Somatic Mutations In Cancer. Somatic mutations in human cancer. Curated database based on peer-reviewed literature. ', term_type='database', provenance='')
def_term43 = Terminology(term_name='ClinVar', term_definition='Phenotype database. Reports relationships between human variations and phentypes. Represents medical phenotypes using ontologies such as  SNOMED CT, GeneReviews, Genetic Home Reference, Office of Rare Diseases, MeSH, and OMIM. Variations are reported according to HGVS standards.', term_type='database' ,provenance='NCBI')
def_term44 = Terminology(term_name='GRC37/38', term_definition='NCBI build of the human genome. 0 based coordinate system.', term_type='genome', provenance='NCBI')
def_term45 = Terminology(term_name='hg19/38', term_definition='UCSC build of the human genome. 1-based coordinate system.|', term_type='genome', provenance='UCSC')
def_term46 = Terminology(term_name='OMIM', term_definition='Gene & variant naming system. Human genes and genetic phenotypes. Focus on relationship between phenotype and genotype. Provides comprehensive description of mutations & corresponding clinical descriptions associated with genes.', term_type='nomenclature', provenance='')
def_term47 = Terminology(term_name='ExAC', term_definition='Exome variant database. Dataset includes 60,000 unrelated individuals sequenced as part of various disease and population genetic studies. Data has been reprocesssed through the same pipeline and called jointly to increase consistency.', term_type='database', provenance='')
def_term48 = Terminology(term_name='Gemini', term_definition='Variant database. Draws from ENCODE tracks, UCSC tracks, OMIM, dbSNP, KEGG, and HPRD.', term_type='database', provenance='')
def_term49 = Terminology(term_name='Mutalyzer', term_definition='Variant nomenclature check. Checks variant conforms to HGVS nomenclature.', term_type='nomenclature', provenance='')


desc_1= Workflow_Description(description= "This is the primary next-generation sequencing analytical pipeline for QIMR Berghofer and most other pipelines are based directly upon it or borrow heavily in terms of tools and workflows.", workflow_manager= "wdl executed by Cromwell", cwl_link= "", sample_qc= "snp array + qpure version 1.1, qsignature version 0.1", fastq_qc= "qprofiler", alignment= "bwa-mem, duplicates - picard markduplicates version & samblaster version", bam_qc= "qprofiler, qcoverage", variant_calling= "SNV - GATK HaplotypeCaller + qSNP, indels - GATK HaplotypeCaller version + q3indel", variant_annotation = "snpEFF version", variant_filtering= "As part of snpEFF", variant_qc = "manual verification", verification= "Illumina SNP array for sample verification", reporting ="", workflow_id =workflow_qimr_som)
desc_2= Workflow_Description(description= "This describes the core research pipeline of the seqliner framework. Inputs paired tumour-normal samples with multiple lanes sequenced per individual.", workflow_manager= "Bpipe version X - groovy with in-house extensions for audit/file management", cwl_link= "", sample_qc= "SNP signature of tumour vs normal and samples compared (separate pipeline)", fastq_qc= "cutadapt v1.5, FastQC v0.11.2", alignment="bwa-mem v0.7.10-r789 + local realignment (GATK RealignerTargetCreator + IndelRealign) + BQSR (GATK Base Recalibrator + GATK PrintReads) + Picard MarkDuplicates v1.119", bam_qc= "bedtools -- coverageBed v2.25.0 + GATK Depthofcoverage v3.6", variant_calling= "SNV & indel - vardict v1.4.6, SNV - GATK Mutect2 v3.6, varscan v2.3.7, GATK Haplotypecaller v3.6", variant_annotation = "VEP version 78 with plugins: condel, exac, eva", variant_filtering= "in-house scripts applying strict filtering rules for low-quality reads, discard low quality reads and germline calls", variant_qc = "", verification= "Manual checks during curation (PathOS)", reporting = "Performance summary report generate", workflow_id =workflow_pm_exome)
desc_3= Workflow_Description(description= "This describes a workflow from the clinical arm of seqliner.", workflow_manager= "Bpipe - groovy with in-house extensions for audit/file management", cwl_link= "", sample_qc= "SNP signature of tumour vs normal and samples compared (separate pipeline)", fastq_qc= "cutadapt v1.5, FastQC v0.11.2", alignment= "bwa-mem v0.7.10-r789 + local realignment (GATK RealignerTargetCreator + IndelRealign) + BQSR (GATK Base Recalibrator + GATK PrintReads) + Picard MarkDuplicates v1.119" , bam_qc= "bedtools -- coverageBed v2.25.0 + GATK Depthofcoverage v3.6", variant_calling= "SNV & indel - GATK Mutect1 v (SNV), Indelocator2, varscan v, GATK Haplotypecaller v3.6", variant_annotation = "VEP version 78 with plugins: condel, exac, eva", variant_filtering= "in-house scripts filtering on low-quality reads (discard) and germline only calls", variant_qc = "", verification= "Manual checks during curation (PathOS)", reporting = "", workflow_id =workflow_pm_panel)
desc_4= Workflow_Description(description= "This pipeline is used for both clinical and research projects. Workflow is executed using queue workflow engine. A tracking database stores sample history and provides run/trend analysis. Individuals may be sequenced across several lanes, with a limit of 3 samples processed in a pipeline batch.", workflow_manager= "In-house scripts and coded AGTC v 1.0. Executed on Broad's Queue workflow engine. In-house scripts coded in variety of languages (python, scala...)", cwl_link= "", sample_qc= "Sample mix-up, contamination checks - in-house script", fastq_qc= "bcl2fastq version, adaptor removal part of bwa", alignment= "bwa", bam_qc= "GATK Baserecalibrator, GATK Indel realigner, in-house scripts for coverage, snpSift for Ti/Tv", variant_calling= "GATK HaplotypeCaller version 3.5, GATK UnifiedGenotyper version 2.8", variant_annotation = "In-house tool incorporating VEP version X and ANNOVAR", variant_filtering= "Filters on annotated terms", variant_qc = "GATK VariantRecalibrator, GATK applyRecalibration", verification= "Scheduled performance tests", reporting = "HGSV format following NPAAC guidelines", workflow_id =workflow_qut_som)
desc_5= Workflow_Description(description= "Perl-based workflow manager, which manages job queues and logs/tracks sample information. A mysql database stores this information. The pipeline itself is largely composed of perl-based scripts with XML configurations. All variants are uploaded to an in-house database. Prior to variant annotation, variants are compared against the in-house database. Only variants novel to the database are annotated which improves the efficiency of the annotation stages. In-house scripts remove variants discovered which are outside of exon boundaries and splice-sites.", workflow_manager="perl-based pipeline manager. Pipeline largely composed of XML and perl scripts" , cwl_link= "", sample_qc= "Adaptor checks performed by sequencing provider", fastq_qc= "Checks of exon read depth", alignment="bwa-mem version 0.6.1-r104", bam_qc= "Novosort - markduplicates, samtools version 0.1.18, GATK indel realignment, GATK BQSR", variant_calling= "SNV & Indels: GATK HaplotypeCaller; SV: Lumpy version 0.2.13, delly version 0.7.3", variant_annotation = "Vep version 137, 138, 142; plugin: CADD", variant_filtering= "in-house scripts: exon & splice sites, flags for dbsnp (MAF) and exac frequencies annotated to vcf" , variant_qc = "GATK VQSR version", verification= "", reporting = "SNVs, indels and SV reported as separate tables", workflow_id =workflow_nci)
desc_6= Workflow_Description(description= "Bioinformatics pipeline is composed of two stages - bam processing and variant calling are part of a NextGENe workflow, with default settings mostly maintained. The variant annotation is performed in Alamut. The pipeline is rarely re-configured.", workflow_manager= "Metadata and files managed manually by batch", cwl_link= "", sample_qc= "", fastq_qc= "bcl2fastq2, additional adaptor trimming as part of NextGENe workflow, FastQC", alignment= "NextGENe", bam_qc= "NextGENe", variant_calling= "NextGENe", variant_annotation = "Alamut Batch, Alamut Visual", variant_filtering= "NextGENe & Alamut", variant_qc = "", verification= "Sanger sequencing, sex checks, manual checks with Alamut visual", reporting =, workflow_id = workflow_schn_panel)
desc_7= Workflow_Description(description= "Most the pipeline in executed on the cloud via Illumina Basespace. Most features parallelised by chromosome. Each panel has preset configurations (determined by the manufacturer) which would rarely be modified by the bioinformatician. Basespace has various in-built internal checks.", workflow_manager= "Illumina BaseSpace", cwl_link= "", sample_qc= , fastq_qc= "part of basespace workflow", alignment= "bwa 0.7.7, samtools version 0.1.19-isis-1.0.1", bam_qc= "Picard CollectMetrics version 1.79", variant_calling= "GATK UnifiedGenotyper version 1.6.23-gf0210b3", variant_annotation = "Illumina Annotation Service, version 3.0 Cartagenia version", variant_filtering= , variant_qc = , verification= "Genome in a bottle reference standards. Every run with a quality control sample with expected values control sets.", reporting ="ACMG guidelines followed. Refseq genes reported. HGSV nomenclature. Manual verification with Alamut visualise.", workflow_id =workflow_pathwest_panel)
desc_8= Workflow_Description(description= "A series of in-house scripts execute the workflow. Sequencing the medical exome is commonly used as an alternative to other diagnostic capture panels (e.g. cancer, truesight) with the advantage that multiple 'panels' can be examined and sequenced in a single test. Exchange of panel-based tests for sequencing of the medical exome has not been fully implemented due to differences in cost.", workflow_manager= "XML-based workflow management + in-house scripts", cwl_link= "", sample_qc= "in-house scripts, Illumina Sequence Analysis Viewer, demultiplexing: bcl2fastq v2.17", fastq_qc= "adaptor removal/trimming: cutadapt v1.8.1, FastQC v0.11.5", alignment= "bwa-kit v0.7.12-r1039 - samblaster v0.1.20 - local realignment with GATK suite v2014.4-2 (RealignerTargetCreator + IndelRealign + Base Recalibrator)", bam_qc= "samtools mpileup v1.0 - Picard CollectInsertSizeMetrics - CalculateHsMetrics v1.114", variant_calling= "SNV + indel - GATK suite v2014.4-2 HaplotypeCaller, GATK GenotypeGVCFs", variant_annotation = " VariantStudio, in-house pipeline", variant_filtering= "QUAL, population frequency, gene of interest list, impact (snpEff)", variant_qc = "manual inspection in IGV", verification=  "gender check (in-house script to check ratios of number of reads mapping to chrX/Y, and ratio of heterozygote/homozygote calls on chrX). Sanger confirmation of pathogenic variants", reporting = "Run and sample QC reports are evaluated prior to reporting", workflow_id =workflow_sap_exome)
desc_9= Workflow_Description(description= "Used for germline samples - supporting individual and trio design configurations. Workflow largely follows GATK Best practices.", workflow_manager= "XMl-based wokflow management + in-house scripts", cwl_link= "", sample_qc= "in-house scripts, Illumina Sequence Analysis Viewer, demultiplexing: bcl2fastq v2.17", fastq_qc= "adaptor removal/trimming: cutadapt v1.8.1, FastQC v0.11.5", alignment= "bwa-kit v0.7.12-r1039 - samblaster v0.1.20 - local realignment with GATK suite v2014.4-2 (RealignerTargetCreator + IndelRealign + Base Recalibrator)", bam_qc= "samtools mpileup v1.0 - Picard CollectInsertSizeMetrics - CalculateHsMetrics v1.114", variant_calling= "SNV + indel - GATK suite v2014.4-2 HaplotypeCaller, GATK GenotypeGVCFs", variant_annotation = "VariantStudio, in-house pipeline", variant_filtering= "VariantStudio, in-house pipeline", variant_qc = "manual inspection in IGV, GATK VariantRecalibrator + ApplyRecalibration", verification= "gender check (in-house script to check ratios of number of reads mapping to chrX/Y, and ratio of heterozygote/homozygote calls on chrX). Sanger confirmation of pathogenic variants", reporting = "Run and sample QC reports are evaluated prior to reporting", workflow_id = workflow_sap_panel)
desc_10= Workflow_Description(description= "Part of the KidGen flagship. Samples processed by IMB have already been analysed as singleton (proband only) panels by SCHN and/or VCGS. If known variants are not discovered as part of these workflows, trio analysis is performed at IMB. Essentially, this pipeline aims to discover novel variants." , workflow_manager= "In-house python-based manager", cwl_link= "", sample_qc= "qsignature", fastq_qc= "fastQC v, cutadapt v", alignment= "bwa mem", bam_qc= "Picard MarkDuplicates, GATK baserecalibrator", variant_calling= "SNV, & indels: GATK HaplotypeCaller and RTG; mitochondrial as separate branch", variant_annotation = "snpeff & snpsifT", variant_filtering= "Family-based models for variant filtering post-annotation (RTG)", variant_qc = "Variant recalibration as part of RTG workflow", verification= "sample mix-ups: qsignature; pedigree check: peddy", reporting = "", workflow_id =workflow_imb_trio)
desc_11= Workflow_Description(description= "This is the primary version of Cpipe supported by Melbourne Genomics", workflow_manager= "Bpipe v0.9.8.8 - groovy", cwl_link= "", sample_qc= "FastQC v0.11.5", fastq_qc= , alignment= "bwa-mem v0.7.13 + local realignment (GATK RealignerTargetCreator + IndelRealign) + GATK Base Recalibrator v3.6 + Picard MarkDuplicates v2.6.0", bam_qc= "bedtools -- coverageBed v2.25.0 + GATK Depthofcoverage v3.6 + Picard CollectInsertSizeMetrics v2.6.0", variant_calling= "SNV & indel - GATK HaplotypeCaller v3.6", variant_annotation = "VEP v85 + plugins (dbNSFP v2.9.1, Condel, Grantham)", variant_filtering= "Transcript filters (NM transcripts preferred -- in-house script) + VEP biotype filters + hard filters (AlleleDepth (AD), Depth (DP), allele frequency (AF), Phred (QUAL))", variant_qc = "", verification= "Karyotype check (X/Y coverage) -- in-house script. Pedigree check (under development)", reporting = "Annotated variants uploaded to LOVD+. Stage and batch reports are generated within Cpipe and evaluated prior to curation.", workflow_id = workflow_mb_exome)
desc_12= Workflow_Description(description= "This is the primary next-generation sequencing analytical pipeline for KCCG, at Garvan Institute. This analytical pipeline has been NATA accredited, as part of KCCG’s clinical accreditation for germline sequencing of DNA from blood. The pipeline is implemented on DNAnexus, a platform as a service that runs on AWS. DNAnexus takes care of data management, security, sharing, access, auditing, and instance management. Each component of the pipeline (eg bwa, HaplotypeCaller, ...) is represented as an individual app, which are threaded together using an in house pipeline generator, which uses the DNAnexus API, and sample metadata to configure custom workflows that are flexible to the structure of the given dataset. All code is individually managed in git, and extensively tested as individual components, and end-to-end, via in house a Bamboo continuous integration server." , workflow_manager= "DNA nexus", cwl_link= "", sample_qc= "SampleID (via AGRF, and extracted from WGS BAM)", fastq_qc= "Illumina HAS, bcl2fastq2, FastQC", alignment= "bwa-mem + GATK Indel Realign + GATK Base Recalibration", bam_qc= "Picard WgsMetrics, GATK DepthOfCoverage", variant_calling= "SNV & Indel - GATK HaplotypeCaller v3.3", variant_annotation = "VEP v74; Gemini; Seave with in house databases: OMIM, ClinVar, COSMIC, Orphanet", variant_filtering= "GATK VQSR", variant_qc = "GATK VariantEval. In house VCF report (vs PED file, kinship coefficients, variant quality and number of categories)", verification= "Recorded in Patient Archive", reporting = "Patient Archive + manual", workflow_id = workflow_kccg_germ)
