## add data to session

flagship_kidgen = Flagship(flagship_name='KidGen Renal genetics', flagship_institute=' Queensland health', flagship_lead='Andrew Mallett', flagshipDiseaseType='Kidney')
flagship_neuro = Flagship(flagship_name='Neuromuscular disorders', flagship_institute='University of Western Australia', flagship_lead='Nigel Laing', flagshipDiseaseType='Neuromuscular')
flagship_mito = Flagship(flagship_name='Mitochondrial disorders', flagship_institute='Murdoch Childrens Hospital', flagship_lead='John Christodoulou', flagshipDiseaseType='Mitochondrial')
flagship_som = Flagship(flagship_name='Somatic cancer', flagship_institute='Peter MacCallum Cancer Centre', flagship_lead='Stephen Fox', flagshipDiseaseType='Cancer')
flagship_brain = Flagship(flagship_name='Brain malformations & leukodystrophies', flagship_institute='University of Melbourne', flagship_lead='Ingrid Scheffer', flagshipDiseaseType='Epileptic Encephalopathy')
flagship_enc = Flagship(flagship_name='Epileptic Encephalopathy', flagship_institute='Murdoch Childrens Hospital', flagship_lead='Richard Leventer', flagshipDiseaseType='Brain malformations and leukodystrophies')
flagship_imm = Flagship(flagship_name='Genetic Immunology', flagship_institute='Australian Nataional Univesity', flagship_lead='Richard Leventer', flagshipDiseaseType='Brain malformations and leukodystrophies')
flagship_mgha = Flagship(flagship_name='Melbourne Genomics Health Alliance', flagship_institute='Melbourne Genomics Health Alliance', flagship_lead='Melbourne Genomics Health Alliance', flagshipDiseaseType='null')
flagship_paed = Flagship(flagship_name='Germline ICCon', flagship_institute='University of Queensland', flagship_lead='Robyn Ward & Gillian Mitchell', flagshipDiseaseType='ICCon')
flagship_null = Flagship(flagship_name='Not affiliated with a Flagship', flagship_institute='null', flagship_lead='null', flagshipDiseaseType='null')

institute_imb = Institute(institute_name = 'Institute for Molecular Bioscience', institute_logo = 'null', institute_long= '153.009784', institute_lat= '-27.498094')
institute_kccg = Institute(institute_name = 'Kinghorn Centre for Clinical Genomics', institute_logo = 'null', institute_long= '151.221706', institute_lat= '-33.878596')
institute_vcgs = Institute(institute_name = 'Victorian Clinical Genetic Services', institute_logo = 'null', institute_long= '144.949343', institute_lat= '-37.793749')
institute_mb = Institute(institute_name = 'Melbourne Bioinformatics', institute_logo = 'null', institute_long= '144.962710', institute_lat= '-37.800365')
institute_pm = Institute(institute_name = 'Peter MacCallum Cancer Centre', institute_logo = 'null', institute_long= '144.956371', institute_lat= '-37.800277')
institute_atcg = Institute(institute_name = 'Institute of Health and Biomedical Innovation', institute_logo = 'null', institute_long= '153.013758', institute_lat= '-27.452918')
institute_sap = Institute(institute_name = 'South Australia pathology', institute_logo = 'null', institute_long= '138.599898', institute_lat= '-34.911586')
institute_qimr = Institute(institute_name = 'QIMR Berghofer', institute_logo = 'null', institute_long= '153.027383', institute_lat= '-27.449517')
institute_pathwest = Institute(institute_name = 'Pathwest', institute_logo = 'null', institute_long= '115.866493', institute_lat= '-31.954350')
institute_schn = Institute(institute_name = 'Sydney Childrens Hospital Network', institute_logo = 'null', institute_long= '151.237574', institute_lat= '-33.917409')


pipeline_imb = Pipeline(pipeline_name='IMB Trio', pipeline_provider='Institute for Molecular Bioscience', institute = institute_imb)
pipeline_kccg = Pipeline(pipeline_name='KCCG', pipeline_provider='Kinghorn Centre for Clinical Genomics', institute = institute_kccg)
pipeline_vcgs = Pipeline(pipeline_name='Cpipe', pipeline_provider='Victorian Clinical Genetic Services', institute = institute_vcgs)
pipeline_mb_cpipe = Pipeline(pipeline_name='Cpipe', pipeline_provider='Melbourne Bioinformatics', institute = institute_mb)
pipeline_pm = Pipeline(pipeline_name='Seqliner', pipeline_provider='Peter MacCallum Cancer Centre', institute = institute_pm)
pipeline_mb_seq = Pipeline(pipeline_name='Seqliner', pipeline_provider='Melbourne Bioinformatics', institute = institute_mb)
pipeline_atcg = Pipeline(pipeline_name='ATCG pipeline', pipeline_provider='Institute of Health and Biomedical Innovation', institute = institute_atcg)
pipeline_sap = Pipeline(pipeline_name='SAP pipeline', pipeline_provider='South Australia pathology', institute = institute_sap)
pipeline_qimr = Pipeline(pipeline_name='QIMR pipelines', pipeline_provider='QIMR Berghofer', institute = institute_qimr)


workflow_pm_panel = Workflow(workflow_name = 'PM panel', library_preparation = 'Comprehensive cancer panel version 2', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'yes' , reference_genome = 'GrCh37' , workflow_usage = 'diagnostic', flagship=flagship_som, pipeline=pipeline_pm)
workflow_pm_exome = Workflow(workflow_name = 'PM exome', library_preparation = ' ', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'GrCh37' , workflow_usage = 'research', flagship=flagship_null, pipeline=pipeline_pm)
workflow_mb_exome = Workflow(workflow_name = 'MB exome', library_preparation = 'Nextera', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', flagship=flagship_null, pipeline=pipeline_mb_cpipe)
workflow_vcgs_exome = Workflow(workflow_name = 'VCGS exome', library_preparation = ' ', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', flagship=flagship_mgha, pipeline=pipeline_vcgs)
workflow_vcgs_panel1 = Workflow(workflow_name = 'VCGS panel 1', library_preparation = ' ', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'no' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', flagship=flagship_mito, pipeline=pipeline_vcgs)
workflow_vcgs_panel2 = Workflow(workflow_name = 'VCGS panel 2', library_preparation = ' ', library_layout = 'paired', sequencing_strategy = 'panel', nata_accreditation = 'no' , reference_genome = 'hg19' , workflow_usage = 'diagnostic', flagship=flagship_brain, pipeline=pipeline_vcgs)
workflow_sap_exome = Workflow(workflow_name = 'SAP exome', library_preparation = 'TruSight', library_layout = 'paired', sequencing_strategy = 'WES', nata_accreditation = 'no' , reference_genome = 'GrCh37' , workflow_usage = 'diagnostic', flagship=flagship_null, pipeline=pipeline_sap)
workflow_sap_panel = Workflow(workflow_name = 'SAP panel', library_preparation = '', library_layout = 'paired', sequencing_strategy = 'Panel', nata_accreditation = 'no' , reference_genome = 'GrCh37' , workflow_usage = 'diagnostic', flagship=flagship_neuro, pipeline=pipeline_sap)
workflow_kccg_som = Workflow(workflow_name = 'KCCG somatic', library_preparation = '', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'no' , reference_genome = 'various' , workflow_usage = 'diagnostic', flagship=flagship_null, pipeline=pipeline_kccg)
workflow_kccg_germ = Workflow(workflow_name = 'KCCG germline', library_preparation = '', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'no' , reference_genome = 'various' , workflow_usage = 'diagnostic', flagship=flagship_null, pipeline=pipeline_kccg)
workflow_qimr_som = Workflow(workflow_name = 'QIMR somatic', library_preparation = '', library_layout = 'paired', sequencing_strategy = 'WGS', nata_accreditation = 'in completion' , reference_genome = 'various' , workflow_usage = 'diagnostic', flagship=flagship_null, pipeline=pipeline_qimr)


def_term1 = Terminology(term_name='Base quality score recalibration', term_definition='Quality scores output by sequencer reflect confidence of the machine in calling a given base. Quality scores are based on sequencer algorithms. Variant callers are heavily reliant on quality scores assigned to each base in each read. BSQR applies machine learning to model errors empirically and adjust quality scores. As the majority of variants have been deposited in dbSNP, novel variants are likely to be sequencing errors. Known variants are masked and the quality scores of remaining SNP’s examined in the context of covariates (quality of surrounding bases, nearby mismatches). ', term_type='', provenance='GATK')
def_term2 = Terminology(term_name='Coverage', term_definition='Number of reads that cover a given nucleotide position.', term_type='', provenance='')
def_term3 = Terminology(term_name='Consequence term', term_definition='As defined by the Sequence ontology. Predict the effect of each allele of each variant on the transcript. Provides a score of impact based on position of variant in the transcript (e.g. splice site, UTR etc).', term_type='', provenance='')
def_term4 = Terminology(term_name='Decomposition', term_definition='Breaking variants into multiple records – splitting multi-allelic sites into separate entries', term_type='', provenance='')
def_term5 = Terminology(term_name='De novo mutations', term_definition='Parents are homozygous reference whereas the child is a het (carries a copy of the alternate allele). A high confidence de novos would be sites where GQ >= 20 across trio with AC <4 or AF <0.1%.', term_type='', provenance='')
def_term6 = Terminology(term_name='PCR Duplicates', term_definition='Duplicates are an artefact of the sequencing process and can be defined as reads arising from the same DNA fragment. In order to generate a signal strong enough for the sequencer to detect, fragments need to be amplified using PCR. The lower the amount of starting the material, the more amplification cycles need to be run. They can be identified by reads mapping to the same genomic coordinates. Failure to remove duplicates can lead to skewed allele frequencies. A high rate of duplication can be diagnostic of an enrichment bias. ', term_type='', provenance='')
def_term7 = Terminology(term_name='Genotype refinement', term_definition='Calculate the probability a sample takes on a homref, het or homref genotype using sample PL. Variation observed for individuals within the trio is used to improve likelihood estimates. Confidence level for genotypes uses Mendelian inheritance rules to strengthen predictions. De novo mutations are tagged (VariantAnnotator) using posterior genotype probabilities. ', term_type='', provenance='')
def_term8 = Terminology(term_name='HaplotypeCaller', term_definition='Discovery of germline SNP’s and indels via local re-assembly of haplotypes. Regions of variation are identified and the region is reassembled locally to identify all possible haplotypes. Possible haplotypes are realigned against the reference to calculate most probable variant. Bayes rule is applied using the likelihood of alleles based on the given the reads to determine the most probable genotype. Genotyping algorithm used to detect het vs hom (expects 0, 50, 100% at variant calling sites). NB this is problematic for somatic samples where frequencies are typically skewed (mutation feature, also contamination).', term_type='', provenance='')
def_term9 = Terminology(term_name='Insert', term_definition='Refers to the read between the paired end adaptors. Insert size should be ~ 300-500 bp for paired end sequencing.  For illumina, this would be the difference between the 5’ positions of the two reads.  ', term_type='', provenance='')
def_term10 = Terminology(term_name='Interval padding', term_definition='Convention of shifting indels furthest left whilst keeping the length of the allele constant. Applies to situations where the indel position can be ambiguous (e.g. STR).', term_type='', provenance='')
def_term11 = Terminology(term_name='Left align', term_definition='Convention of shifting indels furthest left whilst keeping the length of the allele constant. Applies to situations where the indel position can be ambiguous (e.g. STR).', term_type='', provenance='')
def_term12 = Terminology(term_name='Local realignment', term_definition='Serves to reduce detection of false SNP’s generated by incorrect mapping around regions containing indels. Read mapping algorithms operate on each read independently. ‘Hidden’ indels can lead to misalignments leading to discovery of false variants. Local realignment examines all reads which span a position to determine the highest-possible scoring consensus. Sites of high variation are targeted.', term_type='')
def_term13 = Terminology(term_name='Mixed', term_definition="Multinucleotide and an indel at a position e.g. Reference = 'ATA', Sample = 'GTCAGT';", term_type='', provenance='')
def_term14 = Terminology(term_name='MNP', term_definition="multinucleotide variant e.g. Reference = 'ATA', Sample = 'GTC'– substitution mutations across consecutive bases.'", term_type='', provenance='')
def_term15 = Terminology(term_name='Normalisation', term_definition='Describes application of both left alignment and parsimony. see XX', term_type='', provenance='')
def_term16 = Terminology(term_name='Parsimony', term_definition='Representation of a variant by as few nucleotides as possible with allele length > 0. This involves trimming bases on either side of the variant.', term_type='', provenance='')
def_term17 = Terminology(term_name='Proband', term_definition='Person with the genetic disorder (usually the child).', term_type='', provenance='')
def_term18 = Terminology(term_name='Refseq accessions', term_definition='Prefix XM are model refseqs. These represent predicted transcripts produced by an automated annotation pipeline and may differ from curated records. NM represent known transcripts with evidence available and curated.', term_type='', provenance='')
def_term18 = Terminology(term_name='SAM file', term_definition='Tab-delimited text file containing aligned sequence data', term_type='', provenance='')
def_term19 = Terminology(term_name='Symbolic', term_definition='Mutation event not well-modelled', term_type='', provenance='')
def_term20 = Terminology(term_name='Variant annotation', term_definition='Process of ascribing biological function/impact to variant calls', term_type='', provenance='')
def_term21 = Terminology(term_name='GT', term_definition='Genotype at this site, encoded 0 for ref and 1 for alt.', term_type='', provenance='')
def_term22 = Terminology(term_name='AD', term_definition='Unfiltered allele depth (counts) – the number of reads at the position, including those that failed variant caller filters.', term_type='', provenance='')
def_term23 = Terminology(term_name='DP', term_definition='filtered allele depth (counts), including only those which passed variant caller filters.', term_type='', provenance='')
def_term24 = Terminology(term_name='PL', term_definition='Normalised Phred-scaled likelihoods – the likelihood of the genotype scaled by quality', term_type='', provenance='')
def_term25 = Terminology(term_name='GQ', term_definition='Quality of the assigned genotype. Calculated as the difference between two most likely genotypes (based on PL). Low GQ indicates low confidence in choosing one genotype over another.', term_type='', provenance='')
def_term26 = Terminology(term_name='Biotype', term_definition='Biotype of a transcript or regulatory feature. GENCODE term.', term_type='', provenance='GENCODE')
def_term27 = Terminology(term_name='MAF', term_definition='Minor allele frequency, refers to the frequency at which the second most common allele occurs in a given population.', term_type='', provenance='')
def_term28 = Terminology(term_name='Raw coverage', term_definition='Percentage of reads that map to reference genome', term_type='', provenance='')
def_term29 = Terminology(term_name='Informative coverage', term_definition='Uniquely mapped reads ', term_type='', provenance='')
def_term30 = Terminology(term_name='Horizontal coverage', term_definition='Target mapped reads on or near target ', term_type='', provenance='')
def_term31 = Terminology(term_name='Depth', term_definition='Read depth. Vertical coverage. Number of times a given base position has been sequenced', term_type='', provenance='')
def_term32 = Terminology(term_name='Depth of coverage', term_definition='Breadth of coverage. Read length * number of reads / haploid genome length. Higher read depth = more reliable consensus and improved accuracy.', term_type='', provenance='')
def_term33 = Terminology(term_name='Transition/transversion ratios', term_definition=' Serves as a general indicator of quality - ratio should remain constant for targeted regions. Comparable values from literature available.', term_type='', provenance='')
def_term34 = Terminology(term_name='Mapping quality score ', term_definition='probability a read is misplaced based on number of distinct regions within a reference assembly software could map the read, no of base differences between the read and candidate locations in the reference sequence. Mapping scores from different algorithms not comparable. Low mapping quality can be indicative of non-specific amplification, capture off target, contaminatio', term_type='', provenance='')
def_term35 = Terminology(term_name='Capture efficiency', term_definition='percentage of targeted bases that are covered by one or more reads.', term_type='', provenance='')
def_term36 = Terminology(term_name='refseq', term_definition='Genes, transcripts, proteins. Annotated set of reference sequences including genomic, transcript, and protein. A subset of annotations from UCSC and Ensembl.', term_type='database', provenance='NCBI')
def_term37 = Terminology(term_name='refseqgene', term_definition='Subset of NCBI, defines genomic sequences to be used as reference standards for well-characterised genes.', term_type='database', provenance='NCBI')
def_term38 = Terminology(term_name='dbSNP', term_definition='SNP variant database. Contains neutral and disease-causing variants', term_type='database', provenance='NCBI')
def_term39 = Terminology(term_name='dbVAR', term_definition='Structural variant database. contains insertions, deletions, duplications, inversions, multinucleotide substitutions, mobile element insertions, translocations, and complex chromosomal rearrangements. See dbSNP for single nucleotide substitutions', term_type='database', provenance='')
def_term40 = Terminology(term_name='1000 genomes', term_definition='Catologue of human variation', term_type='database', provenance='IGSR')
def_term41 = Terminology(term_name='Genome in a bottle', term_definition='Sets of reference genomes. to develop the technical infrastructure (reference standards, reference methods, and reference data) to enable translation of whole human genome sequencing to clinical practice.', term_type='database', provenance='')
def_term42 = Terminology(term_name='COSMIC', term_definition='Catalogue Of Somatic Mutations In Cancer. Somatic mutations in human cancer. Curated database based on peer-reviewed literature. ', term_type='database', provenance='')
def_term43 = Terminology(term_name='ClinVar', term_definition='Phenotype database. Reports relationships between human variations and phentypes. Represents medical phenotypes using ontologies such as  SNOMED CT, GeneReviews, Genetic Home Reference, Office of Rare Diseases, MeSH, and OMIM. Variations are reported according to HGVS standards.', term_type='database' ,provenance='NCBI')
def_term44 = Terminology(term_name='GRC37/38', term_definition='NCBI build of the human genome. 0 based coordinate system.', term_type='genome', provenance='NCBI')
def_term45 = Terminology(term_name='hg19/38', term_definition='UCSC build of the human genome. 1-based coordinate system.|', term_type='genome', provenance='UCSC')
def_term46 = Terminology(term_name='OMIM', term_definition='Gene & variant naming system. Human genes and genetic phenotypes. Focus on relationship between phenotype and genotype. Provides comprehensive description of mutations & corresponding clinical descriptions associated with genes.', term_type='nomenclature', provenance='')
def_term47 = Terminology(term_name='ExAC', term_definition='Exome variant database. Dataset includes 60,000 unrelated individuals sequenced as part of various disease and population genetic studies. Data has been reprocesssed through the same pipeline and called jointly to increase consistency.', term_type='database', provenance='')
def_term48 = Terminology(term_name='Gemini', term_definition='Variant database. Draws from ENCODE tracks, UCSC tracks, OMIM, dbSNP, KEGG, and HPRD.', term_type='database', provenance='')
def_term49 = Terminology(term_name='Mutalyzer', term_definition='Variant nomenclature check. Checks variant conforms to HGVS nomenclature.', term_type='nomenclature', provenance='')
def_term50 = Terminology(term_name='', term_definition='', term_type='', provenance='')
def_term51 = Terminology(term_name='', term_definition='', term_type='', provenance='')
def_term52 = Terminology(term_name='', term_definition='', term_type='', provenance='')
def_term53 = Terminology(term_name='', term_definition='', term_type='', provenance='')
def_term54 = Terminology(term_name='', term_definition='', term_type='', provenance='')
def_term55 = Terminology(term_name='', term_definition='', term_type='', provenance='')
